### PR TITLE
More acceptance test cleanup

### PR DESCRIPTION
**Pull request by dbelcham**
As part of #4457 we moved more tests around and did cleanup on a couple of them. One test was deleted as it was a duplicate.

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/Core/Recoverability/When_configuring_unrecoverable_exception.cs [hunk 1]
```diff
@@ -19,10 +19,10 @@ public void Should_move_to_error_queue_without_retries()
...
                 await Scenario.Define<Context>(ctx => context = ctx)
                     .WithEndpoint<EndpointWithFailingHandler>(b => b
-                        .When((session, ctx) => session.SendLocal(new InitiatingMessage
-                        {
-                            Id = ctx.TestRunId
-                        }))
+                            .When((session, ctx) => session.SendLocal(new InitiatingMessage
+                            {
+                                Id = ctx.TestRunId
+                            }))
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Core/Recoverability/When_configuring_unrecoverable_exception.cs]: formatting looks a bit off here?

File: src/NServiceBus.AcceptanceTests/Core/Recoverability/When_delayed_and_immediate_retries_with_serialization_exception.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using Features;
+    using MessageMutator;
+    using NUnit.Framework;
+
+    public class When_delayed_and_immediate_retries_with_serialization_exception : NServiceBusAcceptanceTest
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Core/Recoverability/When_delayed_and_immediate_retries_with_serialization_exception.cs]: hm, title doesn't seem to really align any more?
How about `When_failing_mutated_message`?

File: src/NServiceBus.AcceptanceTests/Core/Recoverability/When_delayed_and_immediate_retries_with_serialization_exception.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    {
+        [Test]
+        public async Task Should_preserve_the_original_body_for_serialization_exceptions()
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Core/Recoverability/When_delayed_and_immediate_retries_with_serialization_exception.cs]: I think this behavior is not exclusive for serialization exceptions?

File: src/NServiceBus.AcceptanceTests/Core/Recoverability/When_delayed_retries_with_regular_exception.cs [hunk 4]
```diff
@@ -38,19 +38,13 @@ public RetryEndpoint()
                 {
                     var scenarioContext = (Context) context.ScenarioContext;
                     configure.EnableFeature<TimeoutManager>();
-                    configure.Notifications.Errors.MessageSentToErrorQueue += (sender, message) => { scenarioContext.DelayedRetryChecksum = Checksum(message.Body); };
+                    configure.Notifications.Errors.MessageSentToErrorQueue += (sender, message) => { scenarioContext.DelayedRetryBody = message.Body; };
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Core/Recoverability/When_delayed_retries_with_regular_exception.cs]: you could retrieve the body from the `FailedMessages` on the context and remove the notification subscription here I think.

File: src/NServiceBus.AcceptanceTests/Core/Recoverability/When_delayed_retries_with_serialization_exception.cs [hunk 5]
```diff
@@ -18,17 +18,17 @@ public class When_delayed_retries_with_serialization_exception : NServiceBusAcce
...
                     .When(session => session.SendLocal(new MessageToBeRetried()))
                     .DoNotFailOnErrorMessages())
-                .Done(c => c.DelayedRetryChecksum != default(byte))
+                .Done(c => c.FailedMessages.Any())
                 .Run();
 
-            Assert.AreEqual(context.OriginalBodyChecksum, context.DelayedRetryChecksum, "The body of the message sent to delayed retry should be the same as the original message coming off the queue");
+            var errorBody = context.FailedMessages.Single().Value.Single().Body;
+
+            CollectionAssert.AreEqual(context.OriginalBody, errorBody, "The body of the message sent to delayed retry should be the same as the original message coming off the queue");
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Core/Recoverability/When_delayed_retries_with_serialization_exception.cs]: how is this test now different from https://github.com/Particular/NServiceBus/pull/4560/files#diff-1fbe79961d1be87bbf48939ab5ad9901 ?

File: src/NServiceBus.AcceptanceTests/Recoverability/When_custom_policy_always_moves_to_error.cs [hunk 6]
```diff
@@ -9,21 +11,29 @@ namespace NServiceBus.AcceptanceTests.Recoverability.Retries
     public class When_custom_policy_always_moves_to_error : NServiceBusAcceptanceTest
     {
         [Test]
-        public async Task Should_execute_once()
+        public async Task Should_executed_only_once_and_sent_to_error_queue()
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Recoverability/When_custom_policy_always_moves_to_error.cs]: execute, send

File: src/NServiceBus.AcceptanceTests/Recoverability/When_custom_policy_does_single_delayed_retry_before_move_to_error.cs [hunk 7]
```diff
@@ -11,21 +12,28 @@ namespace NServiceBus.AcceptanceTests.Recoverability.Retries
     public class When_custom_policy_does_single_delayed_retry_before_move_to_error : NServiceBusAcceptanceTest
     {
         [Test]
-        public async Task Should_execute_twice()
+        public async Task Should_execute_twice_and_sent_to_error_queue()
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Recoverability/When_custom_policy_does_single_delayed_retry_before_move_to_error.cs]: send

File: src/NServiceBus.AcceptanceTests/Core/Recoverability/When_delayed_retries_with_serialization_exception.cs (hunk 5 above)
Comment (dbelcham) [inline src/NServiceBus.AcceptanceTests/Core/Recoverability/When_delayed_retries_with_serialization_exception.cs]: This test verifies with only Delayed retries configured, the other test verifies with both delayed and immediate retries configured. Other than that both are identical.

**Comment by bording** [comment]
Given how we currently have to ship test changes with NSB releases, is this not something that we should have gotten into the 6.2 release?

**Comment by dbelcham** [comment]
@bording this is just a fraction of the changes. The previous ones all went with 6.2. Not sure I see why some couldn't go in 6.2 and others in 6.3. The exercise is mostly about moving tests in/out of the Core subfolder to properly scope what is sent to downstreams for execution. There is a small amount of cleanup going on too.

File: src/NServiceBus.AcceptanceTests/Core/Recoverability/When_delayed_retries_with_serialization_exception.cs (hunk 5 above)
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Core/Recoverability/When_delayed_retries_with_serialization_exception.cs]: I think we don't need this variety of retry configuration. Let's go with the one which uses both retry options?

**Comment by timbussmann** [comment]
@bording to add what @dbelcham said, there are more PRs expected to come in the time frame of the following weeks.

File: src/NServiceBus.AcceptanceTests/Recoverability/When_custom_policy_always_moves_to_error.cs [hunk 8]
```diff
@@ -9,21 +11,29 @@ namespace NServiceBus.AcceptanceTests.Recoverability.Retries
     public class When_custom_policy_always_moves_to_error : NServiceBusAcceptanceTest
     {
         [Test]
-        public async Task Should_execute_once()
+        public async Task Should_executed_only_once_and_send_to_error_queue()
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Recoverability/When_custom_policy_always_moves_to_error.cs]: shouldn't that be `execute` instead of `executed`?

File: src/NServiceBus.AcceptanceTests/Recoverability/when_immediate_retries_fail.cs [hunk 9]
```diff
@@ -7,7 +7,7 @@
     using Features;
     using NUnit.Framework;
 
-    public class when_immediate_retries_fail : NServiceBusAcceptanceTest
+    public class When_immediate_retries_fail : NServiceBusAcceptanceTest
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Recoverability/when_immediate_retries_fail.cs]: nitpick, but should the file also be renamed?

**Comment by dbelcham** [comment]
@bording talked to @timbussmann and he said that all this needs is your eyes for a merge once it builds successfully

File: src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj [hunk 10]
```diff
@@ -374,4 +373,6 @@
   </Target>
   <Import Project="..\packages\NuGetPackager.0.6.1\build\NuGetPackager.targets" Condition="Exists('..\packages\NuGetPackager.0.6.1\build\NuGetPackager.targets')" />
   <Import Project="..\packages\GitVersionTask.3.6.5\build\dotnet\GitVersionTask.targets" Condition="Exists('..\packages\GitVersionTask.3.6.5\build\dotnet\GitVersionTask.targets')" />
+  <Import Project="..\packages\NuGetPackager.0.6.0\build\NuGetPackager.targets" Condition="Exists('..\packages\NuGetPackager.0.6.0\build\NuGetPackager.targets')" />
```
Comment (bording) [inline src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj]: Looks like something got messed in the rebase, because the lines NuGetPackager.0.6.0 and GitVersionTask.3.6.3 shouldn't be here.

File: src/NServiceBus.AcceptanceTests/Recoverability/When_custom_policy_does_single_delayed_retry_before_move_to_error.cs [hunk 11]
```diff
@@ -11,21 +12,28 @@ namespace NServiceBus.AcceptanceTests.Recoverability.Retries
...
+                        sendOptions.RouteToThisEndpoint();
+                        sendOptions.SetMessageId(messageId);
+                        return bus.Send(new MessageToBeRetried(), sendOptions);
+                    })
                     .DoNotFailOnErrorMessages())
-                .Done(c => c.MessageSentToErrorQueue)
+                .Done(c => c.FailedMessages.Any())
                 .Run();
 
             Assert.AreEqual(context.Count, 2);
```
Comment (bording) [inline src/NServiceBus.AcceptanceTests/Recoverability/When_custom_policy_does_single_delayed_retry_before_move_to_error.cs]: As part of this PR, do we want to ensure all Assert parameters are in the correct order, or leave that for another PR, because this one is incorrect.

Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Recoverability/When_custom_policy_does_single_delayed_retry_before_move_to_error.cs]: I think we can address this in a dedicated PR, will raise one

**Comment by timbussmann** [comment]
@bording the csproj should be fixed, have a look.
